### DISCUSSION TITLE
Add dependency from a git monorepo

### DISCUSSION THREAD

**Discussion by rsodre** [pnpm/pnpm#8194] [Q&A] [answered]
Hello,

There is a monorepo to which I contribute that contains several packages, which I use on my apps.
Every time I add a feature I want to a package, before I can do any deployments, I need to wait for the maintainer to approve my PR, wait for the next release... just then I can add the new dependency on my app and deploy it. That can take weeks...

Or I could just be using my fork as a dependency...
I know I can add a package from a github link, but **not if that link is a monorepo**.

Can pnpm do that?
It should!

**Comment by zkochan** [comment, marked as answer]
This will add a dependency from the `packages/foo` subdirectory of the monorepo:

```
pnpm add github:user/repo#path:packages/foo
```

This feature was released in pnpm [v9.0.0](https://github.com/pnpm/pnpm/releases/tag/v9.0.0).

**Reply by rsodre** [reply]
glad to hear! but I think I already tried that...

The [package](https://github.com/rsodre/dojo.js/tree/restore-burner/packages/create-burner) I'm adding references another workspace package as `workspace:*`, and pnpm fails to resolve it.

And looks like it's ignoring the `restore-burner` branch I'm passing... (because of the `command not found: restore-burner` error)

```sh
$ pnpm --version
9.3.0
$ pnpm add github:rsodre/dojo.js#path:packages/create-burner&restore-burner
[1] 13326
zsh: command not found: restore-burner
 ERR_PNPM_WORKSPACE_PKG_NOT_FOUND  In : "@dojoengine/core@workspace:*" is in the dependencies but no package named "@dojoengine/core" is present in the workspace

This error happened while installing the dependencies of @dojoengine/create-burner@0.7.0-alpha.2

Packages found in the workspace: 
Progress: resolved 64, reused 54, downloaded 0, added 0

[1]  + exit 1     pnpm add github:rsodre/dojo.js#path:packages/create-burner
```

**Reply by rsodre** [reply]
Also... I tried to hard-code the failing dependency, and add the package from the main branch... it adds!

But when building, that imported package is not built! (no dist folder)

```
$ pnpm run build                               

> pistols@0.8.5 build /Users/roger/Dev/Blockchain/Realms/pistols/client
> npx tsc -p tsconfig.json && NODE_OPTIONS='--experimental-wasm-modules' next build

src/lib/dojo/DojoContext.tsx:2:100 - error TS2307: Cannot find module '@dojoengine/create-burner' or its corresponding type declarations.

2 import { BurnerAccount, useBurnerManager, useBurnerWindowObject, usePredeployedWindowObject } from '@dojoengine/create-burner'

$ ls -l node_modules/@dojoengine/create-burner 
lrwxr-xr-x  1 roger  staff  168 Jun 12 14:22 node_modules/@dojoengine/create-burner -> ../.pnpm/@dojoengine+create-burner@https+++codeload.github.com+rsodre+dojo.js+tar.gz+b4cbe12f14329110d_aiywdljhlomztd7gejckafpaky/node_modules/@dojoengine/create-burner

$ ls -l node_modules/@dojoengine/create-burner/
total 40
-rw-r--r--  1 roger  staff  1061 Jan 16 10:55 LICENSE
drwxr-xr-x  3 roger  staff    96 Jun 12 14:22 node_modules
-rw-r--r--  1 roger  staff  1432 Jun 12 13:52 package.json
-rw-r--r--  1 roger  staff  4166 Jan 16 10:55 readme.md
-rw-r--r--  1 roger  staff   180 Jan 16 10:55 tsup.config.ts

```

**Reply by zkochan** [reply]
This is much harder to implement. I don't think it will be implemented soon.

**Reply by rsodre** [reply]
those 3 issues are blockers for me.
what I'm doing now is re-publishing the package on my personal user org, and replacing all the imports, until the new version is released. looks like a dirty solution, but it's easy and works!
thank you!

**Reply by jvllmr** [reply]
I managed to achieve something similar using overrides. You need to override the sub-dependency with its git path.

**Reply by davidfurlong** [reply]
this didn't work for me.

What worked was this:

`pnpm add "github:oliverdowling/Dexie.js#master&path:/libs/dexie-react-hooks"`

(https://github.com/pnpm/pnpm/issues/4765#issuecomment-2657765089)
